classes/author/Collector.inc.php

```diff
@@ -0,0 +1,232 @@
+<?php
+/**
+ * @file classes/author/Collector.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class author
+ *
+ * @brief A helper class to configure a Query Builder to get a collection of announcements
+ */
+
+namespace PKP\author;
+
+use Illuminate\Database\Query\Builder;
+use Illuminate\Support\Facades\DB;
+use PKP\core\interfaces\CollectorInterface;
+use PKP\plugins\HookRegistry;
+
+class Collector implements CollectorInterface
+{
+    public const ORDERBY_SEQUENCE = 'sequence';
+    public const ORDERBY_ID = 'id';
+
+    /** @var string The default orderBy value for authors collector */
+    public $orderBy = self::ORDERBY_SEQUENCE;
+
+    /** @var DAO */
+    public $dao;
+
+    /** @var array|null */
+    public $contextIds = null;
+
+    /** @var array|null */
+    public $publicationIds = null;
+
+    /** @var string get authors with a family name */
+    protected $familyName = null;
+
+    /** @var string get authors with a given name */
+    protected $givenName = null;
+
+    /** @var string get authors with a specified country code */
+    protected $country = null;
+
+    /** @var string get authors with a specified affiliation */
+    protected $affiliation = null;
+
+    /** @var int */
+    public ?int $count;
```

Comment:
Inititalize these to `null`. I'm not sure how strictly it's enforced, but [from the PHP documentation](https://www.php.net/manual/en/language.oop5.properties.php):
> Typed properties must be initialized before accessing, otherwise an Error is thrown.

Reply:
Fixed (strange though - I would think that all nullable variables would have a default value of null)

Reply:
Agreed, it's a little strange!